### PR TITLE
[DO NOT MERGE] Audit conversion tests + method refactor

### PR DESCRIPTION
**Pull request by filipcirtog**
### All Submissions:

* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: internal/translation/audit/conversion_test.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+func TestConversionContructor(t *testing.T) {
```
Comment (josvazg) [inline internal/translation/audit/conversion_test.go:12]:
```suggestion
func TestConversionConstructor(t *testing.T) {
```

Or maybe even better:

```suggestion
func TestNewAuditConfig(t *testing.T) {
```

File: internal/translation/audit/conversion_test.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	}
+	for _, tc := range testCases {
+		t.Run(tc.title, func(t *testing.T) {
+			actualResult := NewAuditConfig(tc.input)
+			assert.Equal(t, tc.expectedOutput, actualResult)
+		})
+	}
+}
+
+func TestToAndFromAtlas(t *testing.T) {
```
Comment (josvazg) [inline internal/translation/audit/conversion_test.go:103]: This seems like a weird mix, I would just have a `TestToAtlas` and a separate `TestFromAtlas` with only internal types or atlas types respectively.

Alternatively one thing you could do is to have a single `TestToAndFromAtlas` table with only the internal type as inputs, then in the test you do `ToAtlas` and the result you pass to `FromAtlas` and finally compare the final result with the original input, which should match.

**Review by josvazg — COMMENTED** [review]
Looking good, just a comment on some test table that seems a bit mixed up

**Comment by josvazg** [comment]
Please rebase on today's latest main to avoid the Go vulnerability issue in the linter:
https://github.com/mongodb/mongodb-atlas-kubernetes/actions/runs/9777360588/job/26991884113

**Comment by filipcirtog** [comment]
A new PR was created for the same issue.

**Pull request closed without merging**